### PR TITLE
TYP: add Shape alias to pandas._typing

### PR DESCRIPTION
**Pull request by arw2019**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

xref https://github.com/pandas-dev/pandas/pull/37024#discussion_r503488032

cc @simonjayhawkins

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -85,6 +86,7 @@
 Label = Optional[Hashable]
 IndexLabel = Union[Label, Sequence[Label]]
 Level = Union[Label, int]
+Shape = Union[int, Tuple[int, ...]]
```
Comment (jorisvandenbossche) [inline pandas/_typing.py:89]: Why the union of int and tuple. Shape isn't always a tuple?

Comment (simonjayhawkins) [reply pandas/_typing.py:89]: I think in most cases numpy accepts an int as a 1-tuple

Comment (arw2019) [reply pandas/_typing.py]: I think you're right, pushed the change just now

(Since it needs to be indexable, it may have to be a tuple?)

**Comment by jreback** [comment]
@jbrockmendel ok ?

File: pandas/core/ops/array_ops.py [hunk 2]
```diff
@@ -427,15 +427,15 @@ def maybe_upcast_datetimelike_array(obj: ArrayLike) -> ArrayLike:
...
+def _maybe_upcast_for_op(obj, shape: Shape):
     """
     Cast non-pandas objects to pandas types to unify behavior of arithmetic
     and comparison operations.
 
     Parameters
     ----------
     obj: object
-    shape : tuple[int]
+    shape : tuple[int] or int
```
Comment (jbrockmendel) [inline pandas/core/ops/array_ops.py:438]: looks like the annotation and docstring dont match?

Comment (arw2019) [reply pandas/core/ops/array_ops.py:438]: Right. Reverting the change to the docstring

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
lgtm. can you merge master and ping on green.

**Comment by arw2019** [comment]
CI green

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**